test/test_other.py

```diff
@@ -11031,6 +11011,29 @@ def check_symbolmap_info(address, func):
     # The name section will not show bar, as it's inlined into main
     check_symbolmap_info(unreachable_addr, '__original_main')
 
+    # 3. Test symbol map on C++ name mangling
+    self.run_process([EMXX, test_file('other/test_symbol_map.cpp'),
+                      '-O1', '--emit-symbol-map', '-o', 'test_symbol_map.js'])
+    self.assertExists('test_symbol_map.js.symbols')
+
+    out_to_js_call_addr = self.get_instr_addr('call\t0', 'test_symbol_map.wasm')
+    unreachable_addr = self.get_instr_addr('unreachable', 'test_symbol_map.wasm')
+
+    def check_cpp_symbolmap_info(address, func):
+      out = self.run_process([emsymbolizer, '--source=symbolmap', '-f', 'test_symbol_map.js.symbols', 'test_symbol_map.wasm', address], stdout=PIPE).stdout
+      self.assertIn(func, out)
+
+    def check_symbol_map_contains(func):
+      out = read_file('test_symbol_map.js.symbols')
+      self.assertIn(func, out)
+
+    check_cpp_symbolmap_info(out_to_js_call_addr, 'Namespace::foo')         # the function name
+    check_cpp_symbolmap_info(out_to_js_call_addr, 'Namespace::ClassA')      # the parameter
+    check_cpp_symbolmap_info(unreachable_addr, 'Namespace::bar')            # the function name
+    check_cpp_symbolmap_info(unreachable_addr, 'Namespace::ClassA')         # the type parameter
+    check_cpp_symbolmap_info(unreachable_addr, 'Namespace::ClassB')         # the parameter
+    check_symbol_map_contains('out_to_js')                                  # the imported function
```

Comment:
Actually I think maybe we should be making test_symbol_map any more complex by including emsymbolizer stuff too.   Can we go back adding this to a new or existing test_emsymboliser_xx test?

Reply:
Sorry for the late reply, have been busy with work this week.

Actually, I first tried to reuse the `test_dwarf.c` file, but since it's used in a lot of other tests, I created a new standalone file instead.

Now, by using `create_file` to generate the source, this test is self-contained and isn't couple with others.